rfc3164/machine.go.rl

```diff
@@ -0,0 +1,204 @@
+package rfc3164
+
+import (
+    "fmt"
+    "time"
+
+    "github.com/influxdata/go-syslog/v2"
+	"github.com/influxdata/go-syslog/v2/common"
+)
+
+var (
+    errPrival         = "expecting a priority value in the range 1-191 or equal to 0 [col %d]"
+    errPri            = "expecting a priority value within angle brackets [col %d]"
+    errTimestamp      = "expecting a Stamp timestamp [col %d]"
+    errHostname       = "expecting an hostname (from 1 to max 255 US-ASCII characters) [col %d]"
+    errTag            = "expecting an alphanumeric tag (max 32 characters) [col %d]"
+    errContentStart   = "expecting a content part starting with a non-alphanumeric character [col %d]"
+    errContent        = "expecting a content part composed by visible characters only [col %d]"
+    errParse          = "parsing error [col %d]"
+)
+
+%%{
+machine rfc3164;
+
+include common "common.rl";
+
+# unsigned alphabet
+alphtype uint8;
+
+action mark {
+    m.pb = m.p
+}
+
+action set_prival {
+    output.priority = uint8(common.UnsafeUTF8DecimalCodePointsToInt(m.text()))
+	output.prioritySet = true
+}
+
+action set_timestamp {
+    if t, e := time.Parse(time.Stamp, string(m.text())); e != nil {
```

Comment:
Yeah this was something I had to do but then I forgot. Thanks for reminding me.

My idea was to create an option for this parser - eg., `WithYear()` so that the user can decide the behavior.

WDYT?

Reply:
I looked around at logstash and there have been a few attempts to make heuristics.
https://github.com/logstash-plugins/logstash-filter-date/pull/4#issue-26908778

Here is another heuristic:
https://github.com/logstash-plugins/logstash-filter-date/issues/51#issuecomment-309866094

This second link has three configurable heuristic strategies.  With your idea of `WithYear()` would be similar to `disambiguate=from_default`

I personally would use `nearest_default` to handle data coming in between Dec and Jan.

Reply:
These strategies are cool.

I just pushed a strategy pattern for year strategies and implemented a simple one as an example (current year) :)

This way other strategies can be easily implemented (and automatically used) later :)